labman/db/tests/test_sql_connection.py

```diff
@@ -0,0 +1,667 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2017-, labman development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from unittest import TestCase, main
+from os import remove, close
+from os.path import exists
+from tempfile import mkstemp
+
+from psycopg2._psycopg import connection
+from psycopg2.extras import DictCursor
+from psycopg2 import connect
+from psycopg2.extensions import TRANSACTION_STATUS_IDLE
+
+
+from labman.db.settings import labman_settings
+from labman.db.sql_connection import SQLConnectionHandler, Transaction, TRN
+
+
+DB_CREATE_TEST_TABLE = """CREATE TABLE pm.test_table (
+    str_column      varchar DEFAULT 'foo' NOT NULL,
+    bool_column     bool DEFAULT True NOT NULL,
+    int_column      bigint NOT NULL);"""
+
+DB_DROP_TEST_TABLE = """DROP TABLE IF EXISTS pm.test_table;"""
+
```

Comment:
i do not see where the actual db is created.  all tests fail when trying to connect as db does not exist.